cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -236,17 +229,48 @@ func MustNew(cfg *config.Config) *Daemon {
 	onIngestionRetry := func(err error, dur time.Duration) {
 		logger.WithError(err).Error("could not run ingestion. Retrying")
 	}
-	maxRetentionWindow := cfg.EventLedgerRetentionWindow
-	if cfg.TransactionLedgerRetentionWindow > maxRetentionWindow {
-		maxRetentionWindow = cfg.TransactionLedgerRetentionWindow
-	} else if cfg.EventLedgerRetentionWindow == 0 && cfg.TransactionLedgerRetentionWindow > ledgerbucketwindow.DefaultEventLedgerRetentionWindow {
-		maxRetentionWindow = ledgerbucketwindow.DefaultEventLedgerRetentionWindow
+
+	// a metric for measuring latency of transaction store operations
+	txDurationMetric := prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace: daemon.MetricsNamespace(), Subsystem: "transactions",
+		Name:       "operation_duration_seconds",
+		Help:       "transaction store operation durations, sliding window = 10m",
+		Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+	},
+		[]string{"operation"},
+	)
+	txCountMetric := prometheus.NewSummary(prometheus.SummaryOpts{
+		Namespace: daemon.MetricsNamespace(), Subsystem: "transactions",
+		Name:       "count",
+		Help:       "count of transactions ingested, sliding window = 10m",
+		Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+	})
+
+	daemon.metricsRegistry.MustRegister(txDurationMetric, txCountMetric)
```

Comment:
these metrics should be defined an encapsulated in the component which uses the metrics (e.g. cmd/soroban-rpc/internal/db/transaction.go )

Reply:
I'm open to this, but then this means passing around the daemon/metrics registry to every single component in the system which seems silly. The way I have it here creates a separation of concerns of the component saying "here's what I can observe" and the daemon saying "here's how I want it observed". The daemon handles all creation (i.e. the type-specific `prometheus.Observer`), configuration (buckets, etc.), registration, etc. while each component just reports observations.